### PR TITLE
fix: reconcile Secret when userdata changes

### PR DESCRIPTION
**Pull request by trutx**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Whenever the `EKSConfig` `.spec.kubeletExtraArgs` field is updated, the eksconfig controller will reconcile the resulting userdata `Secret` and refresh its `.data.value` field value. Otherwise the `Secret` needs to be manually deleted to trigger a LaunchTemplate update.

I'm also improving the unit tests. Currently, since `cluster.Status.InfrastructureReady == false` all tests end [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/bootstrap/eks/controllers/eksconfig_controller.go#L171). With my changes, the `Secret` is effectively created and checked after reconciliation, thus ensuring the entire reconciliation loop was run.

**Which issue(s) this PR fixes** 

Fixes https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2578

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:
```release-note
Update userdata Secret when EKSConfig kubeletExtraArgs are changed
```

### CODE REVIEW

**Comment by randomvariable** [comment]
Can you squash the linter commit?

File: bootstrap/eks/controllers/eksconfig_controller.go [hunk 1]
```diff
@@ -245,37 +246,54 @@ func (r *EKSConfigReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Man
...
 	// it is possible that secret creation happens but the config.Status patches are not applied
-	if err := r.Client.Create(ctx, secret); err != nil {
-		if !apierrors.IsAlreadyExists(err) {
-			return errors.Wrap(err, "failed to create bootstrap data secret for EKSConfig")
+	secret := &corev1.Secret{}
+	if err := r.Client.Get(ctx, client.ObjectKey{
+		Name:      config.Name,
+		Namespace: config.Namespace,
+	}, secret); err != nil {
+		if apierrors.IsNotFound(err) {
```
Comment (richardcase) [inline bootstrap/eks/controllers/eksconfig_controller.go:257]: This feels a little nested.

Do you think we split this down into functions? Perhaps `createBootstrapSecret` and `updateBootstrapSecret` or something similar.

Comment (trutx) [reply bootstrap/eks/controllers/eksconfig_controller.go:257]: Sure. I went a bit code greedy at the expense of readability. Will change that.

**Comment by richardcase** [comment]
@trutx  - thanks for making that change. It looks like there are some failing tests. Any chance you could look at them? Also, you'll need to run `make generate` to silence the verify job failure.

**Comment by trutx** [comment]
Ran `make generate` locally but nothing changed. Also all tests passed in my laptop. Will retry.

**Pull request merged**